File: kotlin-native/performance/KotlinVsSwift/ring/src/AbstractMethodBenchmark.swift
/*
 * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
 * that can be found in the LICENSE file.
 */

import Foundation

class AbstractMethodBenchmark {
    private let arr: [String] = zdf_win
    private let sequence = "абвгдеёжзийклмнопрстуфхцчшщъыьэюя"
    
    private var sequenceMap: [Character: Int] = [:]
    
    init() {
        var i = 0
        for ch in sequence {
            sequenceMap[ch] = i
            i += 1
        }
    }
    
    func sortStrings() -> Set<String> {
        var size = arr.count
        if (Constants.BENCHMARK_SIZE < size) {
            size = Constants.BENCHMARK_SIZE
        }
        let res = Set(arr[0..<size])
        return res
    }
    
    func sortStringsWithComparator() -> Set<String> {
        var res = Set<String>()
        var size = Constants.BENCHMARK_SIZE < arr.count ? Constants.BENCHMARK_SIZE : arr.count
        arr[0..<size].forEach { (member) in
            res.insert(member)
        }
        return res
    }
}

File: kotlin-native/performance/ring/src/commonMain/kotlin/org/jetbrains/ring/AbstractMethodBenchmark.kt
/*
 * Copyright 2010-2017 JetBrains s.r.o.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.jetbrains.ring

/**
 * Created by Mikhail.Glukhikh on 06/03/2015.
 *
 * A benchmark for a single abstract method based on a string comparison
 */

open class AbstractMethodBenchmark {

    private val arr: List<String> = zdf_win
    private val sequence = "абвгдеёжзийклмнопрстуфхцчшщъыьэюя"

    private val sequenceMap = HashMap<Char, Int>()

    init {
        var i = 0;
        for (ch in sequence) {
            sequenceMap[ch] = i++;
        }
    }

    //Benchmark
    fun sortStrings(): Set<String> {
        val res = arr.subList(0, if (BENCHMARK_SIZE < arr.size) BENCHMARK_SIZE else arr.size).toSet()
        return res
    }

    //Benchmark
    fun sortStringsWithComparator(): Set<String> {
        val res = mutableSetOf<String>()
        res.addAll(arr.subList(0, if (BENCHMARK_SIZE < arr.size) BENCHMARK_SIZE else arr.size))
        return res
    }
}

